src/5/Either.test.ts

```diff
@@ -0,0 +1,128 @@
+import { Either, Left, Right } from 'monet';
+
+import { getCustomerPriceFromReponse, parseJSON } from './Either';
+
+describe(parseJSON, () => {
+  it('returns a JSON object if the string contains valid JSON', () => {
+    const json = {
+      firstName: 'Spike',
+      lastName: 'Spiegel',
+    };
+    const jsonString = JSON.stringify(json);
+
+    const parsedJSON = parseJSON(jsonString).right();
+    expect(parsedJSON).toEqual(json);
+  });
+
+  it('returns an error if the JSON is invalid', () => {
+    const jsonString = 'a;sdklfajdsfafkldf { /// ###';
+
+    const error = parseJSON(jsonString).left();
+    expect(error.message).toEqual('Unexpected token a in JSON at position 0');
+  });
+
+  interface StarWarsCharacter {
+    name: string;
+    height: string;
+    mass: string;
+    hairColor: string;
+    skinColor: string;
+    eyeColor: string;
+    birthYear: string;
+    gender: string;
+    homeworld: string;
+    films: string[];
+    species: any[];
+    vehicles: string[];
+    starships: string[];
+    created: Date;
+    edited: Date;
+    url: string;
+  }
+
+  it('returns an Either that isLeft immediately if the value cant be parsed', async () => {
+    const badResponse = { data: '{$$$$$$$}' };
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏not bad for a scruffy-looking nerf herder...